sotopia/benchmark/cli.py

```diff
@@ -0,0 +1,302 @@
+    agent_index = EnvironmentList.get("01HAK34YPB1H1RWXQDASDKHSNS").agent_index
+    assert isinstance(agent_index, list), "agent_index should be a list"
+
+    env_agent_combo_storage_index_list = []
+    for env_id, index in zip(hard_envs, agent_index):
+        for env_agent_combo_storage in env_agent_combo_storage_list:
+            if env_agent_combo_storage.env_id == env_id:
+                env_agent_combo_storage_index_list.append(
+                    (env_agent_combo_storage, index)
+                )
+
+    list_of_env_agent_combo_storage = []
+    for env_agent_combo_storage, index in env_agent_combo_storage_index_list:
+        agent_ids = env_agent_combo_storage.agent_ids
+        env_id = env_agent_combo_storage.env_id
+        if check_existing_episodes(
+            env_id=env_id, agent_ids=agent_ids, models=model_names, index=index, tag=tag
+        ):
+            logging.info(
+                f"Episode for {env_id} with agents {agent_ids} using {list(model_names.values())} already exists"
+            )
+            continue
+        env_profile = EnvironmentProfile.get(env_id)
+        env = ParallelSotopiaEnv(
+            env_profile=env_profile,
+            model_name=model_names["env"],
+            action_order="round-robin",
+            evaluators=[
+                RuleBasedTerminatedEvaluator(max_turn_number=20, max_stale_turn=2),
+            ],
+            terminal_evaluators=[
+                ReachGoalLLMEvaluator(model_names["env"]),
+            ],
+        )
+        agent_profiles = [AgentProfile.get(id) for id in agent_ids]
+        # make sure the second agent (i.e., the agent being benchmarked) is always the indexed agent
+        agents = [
+            LLMAgent(agent_profile=agent_profile, model_name=agent_model)
+            for agent_profile, agent_model in zip(
+                agent_profiles,
+                [model_names["test_model"], model_names["partner_model"]]
+                if index == "0"
+                else [model_names["partner_model"], model_names["test_model"]],
+            )
+        ]
+        list_of_env_agent_combo_storage.append((env, agents))
+    return list_of_env_agent_combo_storage  # type: ignore
+
+
+def run_async_benchmark_in_batch(
+    *,
+    batch_size: int = 1,
+    model_names: dict[str, LLM_Name] = {
+        "env": "gpt-4",
+        "test_model": "gpt-3.5-turbo",
+        "partner_model": "gpt-3.5-turbo",
+    },
+    tag: str | None = None,
+    push_to_db: bool = False,
+    verbose: bool = False,
+) -> None:
+    url = "https://huggingface.co/datasets/cmu-lti/sotopia/resolve/main/benchmark_agents.json?raw=true"
+    response = requests.get(url)
+    # Check if the request was successful
+    if response.status_code == 200:
+        data = response.json()
+        print("Data fetched successfully")
+    else:
+        print(f"Failed to fetch data. Status code: {response.status_code}")
+    benchmark_combo = initilize_benchmark_combo(data)
```

Comment:
Vulnerability: data is uninitialized if not fetched successfully.

Reply:
raise Exception when response != 200?

Reply:
What's the schema of data?

Reply:
check the initialize function

Reply:
why vulnerable? data won't be there if not fetched successfully

Reply:
```python
    url = "https://huggingface.co/datasets/cmu-lti/sotopia/resolve/main/benchmark_agents.json?raw=true"
    response = requests.get(url)
    # Check if the request was successful
    if response.status_code == 200:
        data = response.json()
        print("Data fetched successfully")
    else:
        print(f"Failed to fetch data. Status code: {response.status_code}")
```

Local variable `data` is not in the scope if it goes into the second branch.

Reply:
hmmmm, then the program would fail? and people could check?

Reply:
No I meant you should raise an explicit exception instead of `print`.

Reply:
raised! :)